site/en/blog/soft-navigations-experiment/index.md

```diff
@@ -0,0 +1,258 @@
+
+## How can I measure Core Web Vitals per soft navigation?
+
+Some considerations need to be taken into account when attempting to measure Core Web Vitals for soft navigations.
+
+Some metrics have traditionally been measured throughout the life of the page: LCP, for example, can change until an interaction occurs. CLS, FID and INP can be updated until the page is navigated away from. Therefore each “navigation” (including the original navigation) will need to finalize these metrics as each new soft navigation occurs.
+
+Similarly, when starting to measure the metrics for the new soft navigation of these long-lived metrics, metrics will need to be “reset” or “reinitialized” and treated as new metrics, with no memory of the values that were set for previous “pages”.
+
+Timings will still be returned in respect of the original “hard” navigation start time. So to calculate LCP for a soft navigation for example, you will need to take the LCP timing and subtract the appropriate soft navigation start time to get a timing relative to the soft navigation. This has been done to keep all timings consistent so other performance timings (such as event timings) that are not measured from the “navigation” time can be treated the same.
+
+### Reporting soft navigations
+
+Once the soft navigations experiment is enabled, the metrics will be reporting via the [`PerformanceObserver`](https://developer.mozilla.org/docs/Web/API/PerformanceObserver) API as usual, but with an additional `navigationId` added to each metric, corresponding to the navigation entry the metric was emitted for.
+
+{% Aside 'note' %}
+Note: [FID is not currently reported](https://bugs.chromium.org/p/chromium/issues/detail?id=1407656) for soft navigations.
+{% endAside %}
+
+You can use a `PerformanceObserver` to observe soft navigations. Below is an example code snippet that logs soft navigation entries to the console:
+
+```js
+const observer = new PerformanceObserver(console.log);
+observer.observe({ entryTypes: ["soft-navigation"] });
+```
+
+This can be used to finalize full-life page metrics for the previous navigation.
+
+### Reporting the metrics against the appropriate URL
+
+As soft navigations can only be seen after they have occurred, some metrics will need to be finalized upon this event, and then reported for the previous URL, as the current URL will now reflect the updated URL for the new page.
+
+The `navigationID` attribute of the appropriate `PerformanceEntry` can be used to tie the event back to the correct URL. This can be looked up with the [`PerformanceEntry API`](https://developer.mozilla.org/docs/Web/API/PerformanceEntry):
```

Comment:
```suggestion
The `navigationID` attribute of the appropriate `PerformanceEntry` can be used to tie the event back to the correct URL. This can be looked up with the [`PerformanceEntry` API](https://developer.mozilla.org/docs/Web/API/PerformanceEntry):
```